include/deal.II/base/mpi.h

```diff
@@ -708,6 +709,416 @@ namespace Utilities
            const T &          object_to_send,
            const unsigned int root_process = 0);
 
+    /**
+     * An interface to be able to use the ConsensusAlgorithm classes. Main
+     * functionality of the implementations is to return a list of process ranks
+     * this process want data from and to deal with the optional payload of the
+     * messages send/received by the ConsensusAlgorithm classes.
+     *
+     * There are two kind of messages:
+     * - send/request message: A message consisting of a data request
+     *   which should be answered by an other process. This message is
+     *   considered as a request message by the receiving rank.
+     * - recv message: The answer to a send/request message.
+     *
+     * @note Since the payloads of the messages are optional, the user has
+     *       to deal with buffers himself. The ConsensusAlgorithm classes 1)
+     *       provides you only with references to empty vectors (of size 0)
+     *       to where the data to be sent can be inserted to or read from and
+     *       2) communicates these vectors blindly.
+     *
+     * @author Peter Munch, 2019
+     */
+    template <typename T1, typename T2>
```

Comment:
Please document the two template types in the class description 4 lines above, e.g. via `@tparam T1 ...`.